src/connection.rs

```diff
@@ -137,14 +135,17 @@ impl<H> Connection<H>
 
     #[cfg(feature="ssl")]
     pub fn encrypt(&mut self) -> Result<()> {
+        let socket = try!(self.socket().try_clone());
         let ssl_stream = match self.endpoint {
-            Server => try!(SslStream::accept(
-                try!(self.handler.build_ssl()),
-                try!(self.socket().try_clone()))),
-
-            Client => try!(SslStream::connect(
-                try!(self.handler.build_ssl()),
-                try!(self.socket().try_clone()))),
+            Server => {
+                let acceptor = try!(self.handler.build_ssl_server());
```

Comment:
A single connector/acceptor can be reused, so it may make more sense to build it once and save it off.